packages/components/src/presentation-components/properties/inputs/AsyncSelect.tsx

```diff
@@ -77,10 +77,16 @@ function NoOptionsMessage<TOption, IsMulti extends boolean = boolean>({ children
 }
 
 function ValueContainer<TOption, IsMulti extends boolean = boolean>({ children, ...props }: ValueContainerProps<TOption, IsMulti>) {
+  const childrenArray = Children.toArray(children);
+  const values = childrenArray.slice(0, -1);
+  const input = childrenArray[childrenArray.length - 1];
   return (
-    <TagContainer {...props.innerProps} className="presentation-async-select-values-container" as={"div"}>
-      {children}
-    </TagContainer>
+    <div className="presentation-async-select-values-container">
+      {input}
+      <TagContainer {...props.innerProps} className="presentation-async-select-tag-container" overflow="truncate" as={"div"}>
+        {props.selectProps.menuIsOpen && props.selectProps.inputValue.length !== 0 ? undefined : values}
```

Comment:
I think we should avoid rendering whole `TagContainer` if we are not rendering values.

Reply:
Changed to no longer render the `TagContainer` when not required.